app/controllers/ems_configuration_controller.rb

```diff
@@ -0,0 +1,106 @@
+class EmsConfigurationController < ApplicationController
+  include Mixins::GenericListMixin
+  include Mixins::GenericShowMixin
+  include Mixins::GenericSessionMixin
+  include Mixins::BreadcrumbsMixin
+  include Mixins::GenericFormMixin
+  include Mixins::ManagerControllerMixin
+  include Mixins::FindRecord
+
+  before_action :check_privileges
+  before_action :get_session_data
+  after_action :cleanup_action
+  after_action :set_session_data
+
+  def self.model
+    ManageIQ::Providers::ConfigurationManager
+  end
+
+  def self.table_name
+    @table_name ||= "ems_configuration"
+  end
+
+  def self.display_methods
+    %w[configuration_profiles configured_systems]
+  end
+
+  def button
+    @edit = session[:edit] # Restore @edit for adv search box
+    params[:display] = @display if display_methods.include?(@display) # Were we displaying nested list
+
+    # Handle Toolbar Policy Tag Button
+    @refresh_div = "main_div" # Default div for button.rjs to refresh
+    model = self.class.model
+    tag(model) if params[:pressed] == "#{params[:controller]}_tag"
+    return if ["#{params[:controller]}_tag"].include?(params[:pressed]) && @flash_array.nil? # Tag screen showing
+
+    if params[:pressed].starts_with?("configured_system_") # Handle buttons from sub-items screen
+      tag(ConfiguredSystem) if params[:pressed] == "configured_system_tag"
+      provision if params[:pressed] == "configured_system_provision"
+    end
+
+    case params[:pressed]
+    when 'ems_configuration_edit_provider'
+      edit
+    when 'ems_configuration_add_provider'
+      new
+    when "ems_configuration_refresh_provider"
+      refresh
+    when "ems_configuration_delete_provider"
+      delete
+    end
+
+    if single_delete_test
+      single_delete_redirect
+    elsif (params[:pressed].ends_with?("_edit_provider") || params[:pressed] == "ems_configuration_add_provider") && @flash_array.nil?
+      if @flash_array
+        show_list
+        replace_gtl_main_div
+      else
+        javascript_redirect(:action => @refresh_partial, :id => @redirect_id)
+      end
+    elsif @refresh_div == "main_div" && @lastaction == "show_list"
+      replace_gtl_main_div
+    else
+      render_flash unless performed?
+    end
+  end
+
+  private
+
+  def self.model_to_name(_provmodel)
+    Dictionary.gettext('ems_configuration', :type => :ui_title, :translate => false)
+  end
+
+  def manager_prefix
+    "configuration_manager"
+  end
+
+  def privilege_prefix
+    "ems_configuration"
+  end
+
+  def refresh
+    assert_privileges("ems_configuration_refresh_provider")
+    manager_button_operation('refresh_ems', _('Refresh'))
+  end
+
+  def concrete_model
+    ManageIQ::Providers::ConfigurationManager
+  end
+
+  def provider_class
+    ManageIQ::Providers::Foreman::Provider
```

Comment:
Why is Foreman still here?

Reply:
@Fryguy This is still here because currently UI does not have a  type drop down in Add a new Provider form, i was hoping once we have new Data driven Provider form in place we will have a type drop down for user to select what type of Provider they want to add with their respective fields. Do you think i should i add a Type drop down on the Add form, if yes, what should be the options, then we can change this code to use value from the type field.

Reply:
How do we do it for other providers?

Reply:
Or is this related to the problem with the Provider model vs ExtManagementSystem model?

Reply:
other provider have a type field on the form, based upon the type selected a Provider object is created. I am working on adding a `Type` field to the Add Configuration Manager form with similar logic that we have for other providers. Should get that in shortly.

Reply:
@Fryguy I am running into an issue with type form field, I have code implemented the same way as it is in other providers.

I am using `EmsConfiguration.supported_types_and_descriptions_hash` to build Type drop down on Configuration Manager Add form, that returns me `cam_configuration` and `foreman_configuration`
but trying to save the new record it throws an error `Error caught: [ActiveModel::MissingAttributeError] can't write unknown attribute "url"`. 
When Add button is pressed in UI, I am creating a new Provider object `EmsConfiguration.model_from_emstype("foreman_configuration").new` that returns me an object of type `ManageIQ::Providers::Foreman::ConfigurationManager ` which does not have url attribute, before my changes we had this part hardcoded to create an object of type `ManageIQ::Providers::Foreman::Provider` 
Please advice.

Reply:
I was able to resolve the issue i mentioned above by adding `supported_types_and_descriptions_hash` method to Provider class but i am not sure if that's the correct fix
```
  def self.supported_types_and_descriptions_hash
    supported_subclasses.each_with_object({}) do |klass, hash|
      if Vmdb::PermissionStores.instance.supported_ems_type?(klass.ems_type)
        hash[klass.ems_type] = klass.description
      end
    end
  end
```

Reply:
@agrare ^?

What do you think?  (we need some sort of Providers permissions I think, or perhaps a way to get the permitted managers, and then from that get their Providers?)

Reply:
Particularly for something like Foreman where neither Manager is the "primary" then permissions on the Provider level are the only way I think.

Reply:
@agrare @Fryguy can you guys help with this change, I am not sure I understand where the permissions need to be set or what other change needs to be made in models.

Reply:
Is there a point to all this?

Consider that for now Foreman::Provider is precisely the class this controller needs.

None of this code should exists in the first place, aren't we just wasting time that could be spent on finishing the proper way of doing that via the API?

Reply:
@Fryguy I agree with @himdel that's the the reason i did not change Add new provider form in this PR. Even if we decide to make changes to existing form , i would prefer a separate follow up PR, this PR is getting way too big.

Reply:
@h-kataria happy to help add permissions and supported_for_create to the provider, would rather that than hard coding Foreman

Reply:
If these are meant to be temporary, then I'm fine with that, especially as the goal is to move towards pluggable DDF for this feature.

> aren't we just wasting time that could be spent on finishing the proper way of doing that via the API?

Isn't that what we are doing with this discussion? It's clear there's a general design
issue with respect to provider/manager split, so discussing it is something we need to do
anyway for the API. It helps to have that discussion in context with the code that is
actually consuming it, hence me bringing in @agrare, as he has some opinions on how best
to deal with the provider/manager split.
Also this thread started as me noticing Foreman in the code when the purpose of the PR is
to remove Foreman from the code, so it stood out. It's not clear from the code that the
purpose for the method is to drive the drop-down for adding a new provider.

Reply:
line of code in question does not drive the drop-down as currently there is no Type drop-down in the UI. Currently users can only add a Foreman Provider and this line is providing class for the new record, when the Add button is UI is pressed.